apps/resources/metac.json

```diff
@@ -4,5 +4,8 @@
   ],
   "dependencies": [
     "org.scalameta:::metac:latest.stable"
-  ]
+  ],
+  "properties": {
```

Comment:
Out of curiosity @alexarchambault, now that this is here, if someone does a `cs fetch metac --json-output-file metac.json`, should they expect to see these properties? I just checked and it isn't but I did a presentation yesterday on Coursier at work and ironically one of the questions that someone asked is how do you see if any properties are being set when you install an app via `cs`. How would you answer that question, and is there any way for a user to see that my chance?

Reply:
You won't see the Java properties through `cs fetch`. They are only used to launch the app, not to fetch it. A way to see them would be generate a bootstrap, and look inside it, like
```bash
$ cs bootstrap metac -o metac --ttl 0
Wrote …/metac
```
(TTL 0 to be sure we fetch the latest app descriptor.)

We can look at the bash preamble of the launcher, to know which command starts the app. We won't find the Java properties there, but we would find [java options](https://github.com/coursier/apps/blob/009379b1716fde82e0f1bcf493d9c26cef5228ee/apps/resources/metals-emacs.json#L8):
```bash
$ head -25 metac
…
exec java $JAVA_OPTS "$@"
```

The properties are set by the launcher right before loading / starting the app, via a property file in the launcher:
```bash
$ unzip -q -c metac coursier/bootstrap/launcher/bootstrap.properties 2>/dev/null
bootstrap.mainClass=scala.meta.cli.Metac
scala.usejavacp=true
```
Using a property file rather than passing the properties via `-D` in the preamble script allows to circumvent any escaping issue. That also works fine on Windows, and allows to reference other properties via [property substitution](https://github.com/coursier/apps/blob/009379b1716fde82e0f1bcf493d9c26cef5228ee/apps/resources/mill.json#L11-L12).

Reply:
Actually, a simpler way would be to pass `-v` to `cs launch`:
```bash
$ cs launch metac -v
  Dependencies:
org.scalameta:metac_2.13.1:latest.stable:
Launching
Running "java" "-Dmetac.version=4.3.8" "-Dscala.usejavacp=true" "-cp" "~/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scalameta/metac_2.13.1/4.3.8/metac_2.13.1-4.3.8.jar:~/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-compiler/2.13.1/scala-compiler-2.13.1.jar:~/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.1/scala-library-2.13.1.jar:~/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scalameta/semanticdb-scalac_2.13.1/4.3.8/semanticdb-scalac_2.13.1-4.3.8.jar:~/.coursier/cache/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-reflect/2.13.1/scala-reflect-2.13.1.jar:~/.coursier/cache/v1/https/repo1.maven.org/maven2/jline/jline/2.14.6/jline-2.14.6.jar:~/.coursier/cache/v1/https/repo1.maven.org/maven2/org/fusesource/jansi/jansi/1.12/jansi-1.12.jar" "scala.meta.cli.Metac"
Usage: scalac <options> <source files>
…
```
We see them as `-D` options.

Reply:
ahhh ok, clever. Cool, the `-v` flag on launch is exactly what I was looking for. Thanks!